client/wsclient.go

```diff
@@ -57,7 +57,7 @@ func (c *wsClient) Start(ctx context.Context, settings types.StartSettings) erro
 	}
 
 	// Prepare connection settings.
-	c.dialer = *websocket.DefaultDialer
+	c.dialer = websocket.DefaultDialer
```

Comment:
We modify `c.dialer` fields later in the code. That would modify `websocket.DefaultDialer` which is not what we want.

Reply:
We don't modify slices. We modify a bool and a pointer field, which is safe. See the code later.

If there is an easy way to make deep copy that would be better and more future-proof, I agree, but for now the code is safe as long as `c.dialer` does not point to `websocket.DefaultDialer`.

Reply:
OK, it's fair to say this is outside the scope of the changeset, and I'll revert this :+1: